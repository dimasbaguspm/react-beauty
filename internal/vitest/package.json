{
  "name": "@react-beauty/vitest",
  "version": "0.0.0",
  "private": true,
  "license": "MIT",
  "type": "module",
  "main": "./src/index.ts",
  "bin": "../../node_modules/.bin/vitest",
  "exports": {
    ".": "./src/index.ts",
    "./selector": "./src/selector.ts"
  },
  "scripts": {
    "lint": "eslint src",
    "check": "tsc --noEmit"
  },
  "devDependencies": {
    "@react-beauty/eslint": "workspace:*",
    "@react-beauty/tsc": "workspace:*",
    "@testing-library/dom": "10.4.0",
    "@testing-library/jest-dom": "6.6.3",
    "@testing-library/react": "16.3.0",
    "jsdom": "26.1.0",
    "vitest": "3.1.2"
  }
}
