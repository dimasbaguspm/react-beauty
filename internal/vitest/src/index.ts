import { resolve } from 'node:path';
import { fileURLToPath } from 'node:url';

import { svgr } from '@react-beauty/vite/app';

import type { Plugin, ViteUserConfig } from 'vitest/config';

const rootDir = fileURLToPath(new URL('.', import.meta.url));

export const vitestConfig = (
  config: ViteUserConfig['test'],
): ViteUserConfig => ({
  plugins: [svgr() as Plugin],
  test: {
    globals: true,
    environment: 'jsdom',
    include: ['src/**/*.{test,spec}.{js,mjs,cjs,ts,mts,cts,jsx,tsx}'],
    setupFiles: [resolve(rootDir, 'setup-file.ts')],
    passWithNoTests: true,
    ...config,
  },
});

export * from 'vitest/config';
