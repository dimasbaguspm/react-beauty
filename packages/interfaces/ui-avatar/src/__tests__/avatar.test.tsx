import { render } from '@react-beauty/vitest/selector';

import { Avatar } from '..';

describe('Avatar', () => {
  it('should work as expected', () => {
    expect(
      render(
        <Avatar size="2xl">
          <Avatar.Initial>dm</Avatar.Initial>
        </Avatar>,
      ).asFragment(),
    ).toMatchSnapshot();
  });

  it('should work as expected', () => {
    expect(
      render(
        <Avatar size="2xl">
          <Avatar.Picture src="https://placecats.com/neo_2/300/200" />
        </Avatar>,
      ).asFragment(),
    ).toMatchSnapshot();
  });
});
