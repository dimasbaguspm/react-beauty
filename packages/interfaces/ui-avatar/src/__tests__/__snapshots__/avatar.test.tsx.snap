// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Avatar > should work as expected 1`] = `
<DocumentFragment>
  <div
    data-testid="avatar-container"
    style="display: inline-block;"
  >
    <div
      class="mocked-styled-0"
      data-size="2xl"
      data-testid="avatar-initial"
    >
      dm
    </div>
  </div>
</DocumentFragment>
`;

exports[`Avatar > should work as expected 2`] = `
<DocumentFragment>
  <div
    data-testid="avatar-container"
    style="display: inline-block;"
  >
    <img
      class="mocked-styled-1"
      data-size="2xl"
      data-testid="avatar-picture"
      src="https://placecats.com/neo_2/300/200"
    />
  </div>
</DocumentFragment>
`;
